certbot/tests/util.py

```diff
@@ -154,19 +155,28 @@ def make_lineage(config_dir, testfile):
     return conf_path
 
 
-def patch_get_utility(target='zope.component.getUtility'):
+def patch_get_utility(target='zope.component.getUtility',
+                        stdout_notification=False):
```

Comment:
I appreciate you factoring this out allowing for potential code reuse in the future, however, I'm not sure if we should take this approach. First, this parameter only affects the `notification` method. Are we going to add similar parameters for other methods in the future as necessary? Every `IDisplay` method prints something to stdout.

Second, while we could remove this parameter and always do this for every `IDisplay` method or have this flag control the behavior of all methods, I'm not sure this is necessary. There is only one place in the entire code base where we currently use this functionality and it is seems like a lot of machinery for this one use case. It also reminds me of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

My personal preference here is to revert these changes and handle the one use case using the approach I described there.